### PR TITLE
Issue #2617270: Move the route subscriber to route provider

### PR DESCRIPTION
**Pull request by mglaman**

### CODE REVIEW

**Comment by pcambra** [comment]
Looks great to me :)

File: src/ProfileHtmlRouteProvider.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Provides HTML routes for the profile entity type.
+ */
+class ProfileHtmlRouteProvider extends DefaultHtmlRouteProvider {
+
+  public function getRoutes(EntityTypeInterface $entity_type) {
+    $collection = parent::getRoutes($entity_type);
+
+    foreach (\Drupal::entityTypeManager()->getStorage('profile_type')->loadMultiple() as $profile_type_id => $profile_type) {
```
Comment (bojanz) [inline src/ProfileHtmlRouteProvider.php]: You have the entity manager injected ($this->entityManager), let's just use it.

File: src/ProfileHtmlRouteProvider.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      $route = new Route(
+        "/user/{user}/edit/user_profile_form/{profile_type}",
+        ['_controller' => '\Drupal\profile\Controller\ProfileController::userProfileForm'],
+        ['_profile_access_check' =>  'add'],
+        ['parameters' => [
+          'user' => ['type' => 'entity:user'],
+          'profile_type' => ['type' => 'entity:profile_type'],
+          $profile_type_id => ['type' => 'profile:' . $profile_type_id],
```
Comment (bojanz) [inline src/ProfileHtmlRouteProvider.php]: What's this ($profile_type_id as a param) about? It's not actually used by userProfileForm.

Comment (pcambra) [inline src/ProfileHtmlRouteProvider.php]: That is coming from my change in the subscriber and I'd like to know it too, seems to be a pattern for routes and if I remember correctly, required for the routes to work.

Comment (mglaman) [inline src/ProfileHtmlRouteProvider.php]: It's used to upcast the param into a loaded object by \Drupal\Core\ParamConverter\EntityConverter. If my recent deep-dive into D8's routing is correct

**Comment by mglaman** [comment]
@bojanz I updated usaged to ````$this->entityManager````

**Pull request merged**